dandischema/datacite.py

```diff
@@ -153,7 +153,7 @@ def to_datacite(
                 continue
 
         contr_all = []
-        if getattr(contr_el, "roleName"):
+        if contr_el.roleName:
```

Comment:
what was rationale for this particular line change? `getattr` is safer and I might have been used for a reason (some prior version having no `.roleName`?) so I would have just kept this as is.

Otherwise looks good to me

Reply:
`getattr()` raises `AttributeError` if the attribute doesn't exist (according to the [`hasattr()` docs](https://docs.python.org/3/library/functions.html#hasattr)), so this change is actually better than calling `getattr()`, because it's a more standard expression of pulling an attribute from an object.

Reply:
I've changed it only because I realize that `roleName` has `None` as default, so the attribute should always exist, but I could revert

Reply:
As I mentioned, I'd keep this the way it is now, since the old and new versions have exactly the same effect, but the new version is more standard.

Reply:
Thank you @waxlamp ! Indeed, this RF is ok here (although unrelated to the purpose of the PR).  I guess while analyzing for it, I have also misguided myself by the fact that `dict.get(key)` does default to not blowing up but just returning `None` if `key` is not in the `dict`.  May be original intention was similarly misguided in 4878d233b7f9b8fbbc81423e8eb8d6645d773f92 @djarecka but lacked that specification of the default?

Reply:
A perhaps little known fact (one that I didn't know until recently myself): `dict.get()` takes a second argument for the default value to return for a missing key (which itself defaults to `None`).

Reply:
@yarikoptic - the only reason why I've changes it here was because I was exploring the default behavior if nothing is passed as `roleName` (that was the example that caused an issue for you).

Reply:
@waxlamp - yes, that's true that `getattr` returns `None`, so it's often a save check that avoids raising any exceptions. Probably that was the reason why I used it at the beginning.

Reply:
> @waxlamp - yes, that's true that `getattr` returns `None`, so it's often a save check that avoids raising any exceptions. Probably that was the reason why I used it at the beginning.

may be a typo (`dict.get`  instead of `getattr`?) since that is the point @waxlamp [mentioned above](https://github.com/dandi/dandischema/pull/81#discussion_r705548634) that `getattr` does raise exception if the default is not specified explicitly (which differs in behavior from `dict.get` where default does not need to be specified and defaults to `None`)

Reply:
overall -- there is no point to use `getattr(obj, "string with the fieldname")` over `obj.fieldname`, unless providing an explicit default, as e.g. `getattr(obj, "string with the fieldname", None)`

Reply:
yes, I'm sorry. I've mixed also in the discussion `getattr` and `dict.get`. But I understand that this could stay like this